src/session.rs

```diff
@@ -295,4 +320,16 @@ impl Session {
     pub async fn close(self) -> Result<(), Error> {
         delegate!(self.0, imp, { imp.close().await })
     }
+
+    /// Leak the underlying ssh multiplex master.
+    /// Return (path to control socket, path to ssh multiplex output log)
+    pub fn leak(self) -> (Box<Path>, Option<Box<Path>>) {
```

Comment:
I think we should avoid calling this "leak", since that has a pretty strong association with leaking memory in Rust (see `Box::leak`). One option is to instead just have a configuration method for "what should happen on drop" — something like `close_on_drop(yes: bool)`, which the user can toggle on or off. If you want to keep it a destructing method like this, I'd go with `fn detach`.

Reply:
I'd like to keep it as it return the path to control socket and the path to the master_log, which is useful for process mux implementation.

Reply:
Done!